### ISSUE TITLE
The panel created by color-prefs:add-to-preferences-panel needs a scroll bar.

### ISSUE THREAD

**Issue by tgbugs** [racket/gui#242]
I have a use case where I am [adding](https://github.com/tgbugs/laundry/blob/next/org/mode.rkt) all the Org mode faces to DrRacket. The color preferences panel does not have a scroll bar, so the result is that the whole preferences window expands to make all the different faces visible at the same time. This causes the bottom of the window to disappear off the bottom of the screen, and affects all other preferences tabs.

I think the solution is to add a scroll bar to the color preferences panel so that it does not cause the whole preferences window to resize.

**Comment by rfindler** [comment]
It would certainly be possible to add an option to `color-prefs:add-to-preferences-panel` to do that. I'm thinking that such a large number of preferences in this dialog is going to be pretty unwieldy, tho.

**Comment by rfindler** [comment]
Is [this commit](https://github.com/racket/gui/commit/5cd210c6505c8b2eaa92bb52566414f68430253b) what you wanted?

**Comment by tgbugs** [comment]
Tested and it works like a charm! One issues is that scrolling on linux is extremely laggy, possibly related to rendering the moving text fields? Not sure.

**Comment by rfindler** [comment]
I believe that's a larger issue without a simple fix, I am sorry to say.
You may wish to consider a GUI that is a little easier to work with; that
may be the best solution.

Robby

**Comment by tgbugs** [comment]
I'm less worried about the performance issues than I am about accidentally destroying the usability of the preferences window.

**Comment by rfindler** [comment]
May I ask why you're adding so many of these configurations? What use-case / group are you helping with this?

**Comment by tgbugs** [comment]
My general objective is to provide some level of feature parity with what is available in the Emacs implementation of Org mode (thus the proliferation of faces). My second objective was to try to follow the existing DrRacket conventions for customization so that the settings can be discoverable.

**Comment by rfindler** [comment]
Oh, I'm not familiar with org mode, but I wonder if there might be a simpler GUI to offer for configuration. I guess I would have to learn more about org mode to make a more useful suggestion.

**Comment by tgbugs** [comment]
I'll explore other options. Using the existing preferences panel was mostly because I was following some of the existing patterns I found in the source. Given the underlying complexity, it almost certainly warrants its own separate configuration GUI.

With regard to this issue, I brought it up primarily because I was worried that any other extension writers might accidentally cause the same window resizing issues that I did and thought it might be good to protect against that, which is something that the style solution does not guarantee.
